src/structarray.jl

```diff
@@ -446,9 +446,8 @@ end
 for op in [:cat, :hcat, :vcat]
     @eval begin
         function Base.$op(args::StructArray...; kwargs...)
-            f = key -> $op((getproperty(t, key) for t in args)...; kwargs...)
             T = mapreduce(eltype, promote_type, args)
-            StructArray{T}(map(f, propertynames(args[1])))
+            StructArray{T}(map((oargs...) -> $op(oargs...; kwargs...), map(components, args)...))
```

Comment:
I agree with the change (iterating on a `Tuple` as it was done before seems incorrect).

This is a little bit hard to read though. Could it be split over several lines giving meaningful names to the variables? Maybe something like

```julia
for op in [:cat, :hcat, :vcat]
    f = Symbol(:curried, op) # or another descriptive name
    @eval begin
        function Base.$op(args::StructArray...; kwargs...)
            $f(vs...) = $op(vs...; kwargs...)
```

From what I understand named functions are a bit better than anonymous functions here for the purpose of error messages (if the anonymous function throws, the stack trace is a bit harder to read).